frontend/src/js/controllers/permissionCtrl.js

```diff
@@ -14,6 +14,29 @@
 
         // message for not verified users
         vm.emailError = utilities.getData('emailError');
+
+        //user email redirect
+        vm.user={};
+
+        var userKey = utilities.getData('userKey');
+
+        var parameters = {};
+        parameters.url = 'auth/user/';
+        parameters.method = 'GET';
+        parameters.token = userKey;
+        parameters.callback = {
+            onSuccess: function(response) {
+                var status = response.status;
+                var result = response.data;
+                vm.user = result;
+            },
+        };
+
+        utilities.sendRequest(parameters);
```

Comment:
I think there is no need to call this API as you are only using username and email that you can get form another controller. try to reuse the service please @Obtin

Reply:
okay :-)

Reply:
@aka-jain username can be accessed from the login controller and then can be accessed in PermCtrl but for email i have to call an api if not in PermCtrl then in  login controller (this api will be called everytime even if the user's email is verified) should i proceed this way? But i think its better to call this api in PermCtrl .

Reply:
@Obtin check web controller which is using 'user' API and loads every information you need it here. The file name is 'webCtrl.js'. Use email id from this file.